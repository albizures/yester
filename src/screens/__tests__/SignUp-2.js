import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import SignUp2 from '../SignUp-2'
const navigation = {
  navigate: jest.fn(),
}

describe('src/screens/SignUp-2.js', () => {
  it('should render', () => {
    const tree = renderer.create(<SignUp2 navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button to free trail is pressed', () => {
    it('should navigate to SignUp screen', () => {
      const shallowSubscription = shallow(<SignUp2 navigation={navigation} />)
      const shallowButton = shallowSubscription.find(Button)
      shallowButton.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.navigate).toHaveBeenCalledWith('LogIn')
    })
  })
})
