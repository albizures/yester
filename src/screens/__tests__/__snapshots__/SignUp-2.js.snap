// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/SignUp-2.js should render 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "marginTop": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    SignUp-2
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        to Create Account
      </Text>
    </View>
  </View>
</View>
`;
